scripts/docs/generate_asset_documentation.py

```diff
@@ -103,7 +103,8 @@ def create_asset_info(asset_config):
         if asset_config.type == AssetType.DATA:
             return DataInfo(asset_config)
```

Comment:
Not related to this PR, but you could create a dict that maps asset type to the function that generates its documentation, which would greatly simplify the above and probably also take over the function of SUPPORTED_ASSET_TYPES. Something like:

```python
TYPE_TO_DOC_FUNCS = {
    AssetType.ENVIRONMENT: EnvironmentInfo,
    AssetType.COMPONENT: ComponentInfo,
    ...
}
```

and then create_asset_info would do something like:

```python
if asset_config.type not in TYPE_TO_DOC_FUNCS:
    warn and return
return TYPE_TO_DOC_FUNCS[asset_config.type](asset_config)
```